### ISSUE TITLE
PreProcessor Support

### ISSUE THREAD

**Issue by con-ssc** [enjalot/blockbuilder#189]
It would be nice if you could add support for pre processed languages like typescript, jsx or sass, stylus and so on.

**Comment by enjalot** [comment]
My current preference is to have people do this within the block by including a client-side preprocessor script. People have done this with babel-standalone for jsx/es2015 http://blockbuilder.org/micahstubbs/9180020ff8265eade016a82ba54db26b

for typescript you could use something like:
https://github.com/niutech/typescript-compile

In tributary I tried to support other preprocessed filetypes like CoffeeScript, but with blockbuilder I'd like to keep it as "native" as possible. The idea is that you should be able to clone the gist made by blockbuilder and run it from a local server and have it work. This wouldn't be possible if we magically preprocessed files for you.

**Comment by enjalot** [comment]
Apparently we will need to fix something to support his fully, this example should work:
http://blockbuilder.org/vasturiano/351c45c55db3f997b8d8ca30447ac0a3

it doesn't because when a user loads a script like `<script type="text/jsx" src="foo.jsx"></script>` we inline the contents of the file into the html for the iframe. when we do it we lose the type="text/jsx" which breaks babel-standalone. This is a bug we can fix to preserve the type when we inline and it should be possible to use any preprocessor that depends on the type.

**Comment by curran** [comment]
@enjalot This might be a good juncture to mention that I'm having success with the sandbox module outside of Blockbuilder. I've released it as [magic-sandbox](https://github.com/curran/magic-sandbox), and it's in use here [https://curran.github.io/d3-in-motion/#2/2/7](https://curran.github.io/d3-in-motion/#2/2/7) (from [curran/d3-in-motion](https://github.com/curran/d3-in-motion)).

One possible direction - we could modify Blockbuilder to depend on `magic-sandbox`, and fix the issue over there. I'm also interested in supporting different languages in the sandbox.

**Comment by con-ssc** [comment]
@enjalot the typescript compile seems a bit deprecated when I read "TypeScript 0.8"

**Comment by enjalot** [comment]
@curran that would be great, I'm all for abstracting that part of the code and making it easier to support with your help :) Would you feel comfortable submitting a PR? Or at least making a new issue specifically about using magic-sandbox?

@con-ssc my point is that whatever solution I would use to support it in blockbuilder could also just be included in the gist, because at the end of the day it requires a browser-based interpreter. If you want to use typescript you'll have to find a solution that works for you.

**Comment by con-ssc** [comment]
@enjalot well, in browser compilation is possible I think just like in jsbin or jsfiddle. I tried it on blockbuilder with systemjs but ran into a problem with http/https while requesting local data

**Comment by micahstubbs** [comment]
will close this in favor of the babel-standalone for now.  

@con-ssc we could re-visit this once #191 **adopt magic-sandbox** is complete
